application/src/main/java/org/togetherjava/tjbot/db/MessageMetadataListener.java

```diff
@@ -0,0 +1,52 @@
+package org.togetherjava.tjbot.db;
+
+import net.dv8tion.jda.api.entities.ChannelType;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.db.generated.tables.MessageMetadata;
+
+/**
+ * Listener responsible for persistence of text message metadata
+ */
+public final class MessageMetadataListener extends ListenerAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(MessageMetadataListener.class);
+
+    private final Database database;
+
+    /**
+     * Creates a new message metadata listener, using the given database.
+     *
+     * @param database the database to store message metadata
+     */
+    public MessageMetadataListener(Database database) {
+        this.database = database;
+    }
+
+    /**
+     * Stores the relevant message metadata for on message received event.
+     * 
+     * @param event incoming message received event.
+     */
+    @Override
+    public void onMessageReceived(MessageReceivedEvent event) {
+        if (!event.getAuthor().isBot() && !event.isWebhookMessage()
+                && !event.isFromType(ChannelType.PRIVATE) && event.isFromType(ChannelType.TEXT)) {
```

Comment:
> !event.isFromType(ChannelType.PRIVATE)
Why listen to a MessageReceivedEvent when you don't want private channels?

Use the GuildMessageReceivedEvent

Reply:
Again, used from one of the started examples. No idea what a GuildMessageReceivedEvent was.
Seems it would eliminate some of the checks so thats a yes to replacement.

Reply:
And again, examples can't show everything possible, look further then your nose.